packages/jest-worker/src/workers/ChildProcessWorker.ts

```diff
@@ -147,6 +153,13 @@ export default class ChildProcessWorker implements WorkerInterface {
     }
   }
 
+  monitorHeartbeat(onExceeded: () => any): void {
+    clearTimeout(this._heartbeatTimeout);
+    this._heartbeatTimeout = setTimeout(() => {
```

Comment:
why a `timeout` and not an `interval`?

Reply:
Maybe my logic is wrong then.

The child processes do `setInterval` to simulate the heartbeat behavior while the worker simply handles the case where there have been no messages from the child process for n-time.

From what I understand it may not be for child processes to fake the heartbeat but directly to the worker ?

![image](https://user-images.githubusercontent.com/32459935/97308583-501e7600-1861-11eb-899b-dfaf16b5c539.png)

Reply:
No I think you're right, Simen probably just read over this very quickly :)

Reply:
ah yes, ignore me 👍